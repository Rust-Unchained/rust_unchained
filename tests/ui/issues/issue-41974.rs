#[derive(Copy, Clone)]
struct Flags;

trait A {
}

impl<T> Drop for T where T: A {
    //~^ ERROR E0120
    fn drop(&mut self) { }
}

fn main() {}
