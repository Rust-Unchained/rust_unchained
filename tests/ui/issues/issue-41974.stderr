error[E0120]: the `Drop` trait may only be implemented for local structs, enums, and unions
  --> $DIR/issue-41974.rs:7:18
   |
LL | impl<T> Drop for T where T: A {
   |                  ^ must be a struct, enum, or union in the current crate

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0120`.
